File: 0x08-python-more_classes/4-rectangle.py
#!/usr/bin/python3
"""This module creates a class Rectangle"""
class Rectangle:
    """This is a class Rectangle"""

    def __init__(self, width=0, height=0):
        """The __init__ method

        Attributes:
            width: the width of the rectangle.
            height: the height of the rectangle
        """
        self.__width = width
        self.__height = height

    @property
    def width(self):
        """This is the width getter method"""
        return self.__width

    @width.setter
    def width(self, value):
        if not isinstance(value, int):
            raise TypeError('width must be an integer')
        if value < 0:
            raise ValueError('width must be >= 0')
        self.__width = value

    @property
    def height(self):
        """This is the height getter method"""
        return self.__height

    @height.setter
    def height(self, value):
        if not isinstance(value, int):
            raise TypeError('height must be an integer')
        if value < 0:
            raise ValueError('height must be >= 0')
        self.__height = value

    def area(self):
        """Returns the rectangle's area"""
        return self.__height * self.__width

    def perimeter(self):
        """Returns the perimeter of the rectangle"""
        if self.__height == 0 or self.__width == 0:
            return 0
        else:
            return 2 * (self.__height + self.__width)

    def __str__(self):
        rect_string = ''
        if self.__height == 0 or self.__width == 0:
            return ''
        else:
            for i in range(0, self.__height):
                rect_string += '#' * self.__width + '\n'
        return rect_string

    def __repr__(self):
        return 'Rectangle(width={}, height={})'.format(
                self.__width, self.__height)

File: 0x13-javascript_objects_scopes_closures/4-rectangle.js
#!/usr/bin/node
class Rectangle {
  constructor (width, height) {
    if (width <= 0 || height <= 0 || !width || !height) {
      return;
    } else {
      this.width = width;
      this.height = height;
    }
  }

  print () {
    let rectangle = '';
    for (let i = 1; i <= this.height; i++) {
      for (let j = 1; j <= this.width; j++) {
        rectangle += 'X';
      }
      if (i !== this.height) {
        rectangle += '\n';
      }
    }
    console.log(rectangle);
  }

  double () {
    this.height = this.height * 2;
    this.width = this.width * 2;
  }

  rotate () {
    let tmp = this.height;
    this.height = this.width;
    this.width = tmp;
  }
}
module.exports = Rectangle;
